app/models/zone.rb

```diff
@@ -261,4 +262,8 @@ def check_zone_in_use_on_destroy
     msg = message_for_invalid_delete
     raise msg if msg
   end
+
+  def clear_my_region_zone_cache
+    MiqRegion.my_region.clear_my_relation_cache
+  end
```

Comment:
Want to point this out.

My hypothesis is we rarely create a zone. And when we do, it is this server (or this test suite) that needs to access that zone for the near future.

But want to highlight that assumption so someone can throw 🍅

Reply:
I think this is true, but hard to tell - have you tried going through the UI and seeing what happens?  Perhaps create a new provider attached to that zone immediately after creating a new zone.  I'm wondering if you'll start getting failures when it tries to put things like the validation on the queue.

You could also do a broadcast, but that's a lot more complicated.